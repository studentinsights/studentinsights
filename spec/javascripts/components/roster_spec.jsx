import SpecSugar from '../support/spec_sugar.jsx';
import Roster from '../../../app/assets/javascripts/components/flexible_roster.jsx';

describe('Roster', function() {
  
  const ReactDOM = window.ReactDOM;

  const helpers = {
    renderInto: function(el, props) {
      ReactDOM.render(<Roster {...props} />, el);
    }
  };

  const props = {
    columns: [
      { label: 'Test Label 1', key: 'element1'},
      { label: 'Test Label 2', key: 'element2'},
      { label: 'Test Label 3', key: 'element3'}
    ],
    rows: [
      { id: 1, element1: 'Data 1-1', element2: 'Data 1-2', element3: 'Data 1-3'},
      { id: 2, element1: 'Data 2-1', element2: 'Data 2-2', element3: 'Data 2-3'},
      { id: 3, element1: 'Data 3-1', element2: 'Data 3-2', element3: 'Data 3-3'}
    ],
    initialSortIndex: 0
  };

  SpecSugar.withTestEl('high-level integration test', function() {
    it('renders the correct headers', function() {
      const el = this.testEl;

      helpers.renderInto(el, props);
      const headers = $(el).find('#roster-header th');

      expect(headers.length).toEqual(3);
      expect(headers[0].innerHTML).toEqual('Test Label 1');
    });

    it('renders the correct data', function() {
      const el = this.testEl;

      helpers.renderInto(el, props);
      const dataElements = $(el).find('#roster-data tr');

      expect(dataElements.length).toEqual(3);

      const firstDataRows = dataElements.eq(0).find('td');
      expect(firstDataRows[0].innerHTML).toEqual('Data 1-1');
    });
  });
});