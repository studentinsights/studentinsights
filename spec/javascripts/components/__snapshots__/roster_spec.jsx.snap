// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roster renders correctly 1`] = `
<div
  className="FlexibleRoster"
>
  <table
    className="roster-table"
    id="roster-table"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <thead>
      <tr
        className="column-groups"
      />
      <tr
        id="roster-header"
      >
        <th
          className="sort-header sort-down"
          onClick={[Function]}
        >
          Test Label 1
        </th>
        <th
          className="sort-header"
          onClick={[Function]}
        >
          Test Label 2
        </th>
        <th
          className="sort-header"
          onClick={[Function]}
        >
          Test Label 3
        </th>
      </tr>
    </thead>
    <tbody
      id="roster-data"
    >
      <tr
        style={
          Object {
            "backgroundColor": "#FFFFFF",
          }
        }
      >
        <td>
          Data 1-1
        </td>
        <td>
          Data 1-2
        </td>
        <td>
          Data 1-3
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "#F7F7F7",
          }
        }
      >
        <td>
          Data 2-1
        </td>
        <td>
          Data 2-2
        </td>
        <td>
          Data 2-3
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "#FFFFFF",
          }
        }
      >
        <td>
          Data 3-1
        </td>
        <td>
          Data 3-2
        </td>
        <td>
          Data 3-3
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
